### PR TITLE
Add Tests and CI

### PR DESCRIPTION
**Pull request by tubbo**
Adds some unit tests and CI configuration so we can make changes without
fear.

### CODE REVIEW

**Comment by meowsus** [comment]
It's... It's beautiful...

Thanks @tubbo. I'll futz around in here tonight.

**Comment by tubbo** [comment]
I did this to prepare the repo for my next PR, which is adding Rugged and using Octokit a bit more...so one doesn’t need hub & git installed to get going.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Tests
```
Comment (tubbo) [inline .github/workflows/tests.yml:1]: hey i'm not sure why the tests didn't run here, are actions enabled on this project?

Comment (meowsus) [reply .github/workflows/tests.yml:1]: They're enabled. Maybe it needs to be merged first? I'll yolo merge and we can clean up if need be.

**Review by meowsus — APPROVED** [review]
YOLOL WTFMLOL

**Pull request merged**